File: q4/q4.py
class Solution:
    def isValid(self, s: str) -> bool:
        new =[]
        check = {']':'[','}':'{',')':'('}
        for char in s:
            if char in check.values():
                new.append(char)
            elif char in check.keys():
                if new==[] or new.pop()!=check[char]:
                    return False
        return new==[]

File: q4/q4.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char> news;
        int i;
        for(i=0;i<s.size();i++)
        {   if(news.empty()||s[i]=='{'|| s[i]=='('|| s[i]=='[')
                news.push(s[i]);
            else
            {   if(s[i]==']'&& news.top()!='[')
                    return false;
                if(s[i]=='}'&& news.top()!='{')
                    return false;
                if(s[i]==')'&& news.top()!='(')
                    return false;
                news.pop();
                
            }
        }
        return news.empty();
    }
};